geniza/corpus/forms.py

```diff
@@ -168,3 +171,41 @@ def clean(self):
             self.add_error(
                 "q", _("Relevance sort is not available without a keyword search term.")
             )
+
+
+class DocumentChoiceField(forms.ModelChoiceField):
+    label_template = get_template("corpus/snippets/document_option_label.html")
+
+    def label_from_instance(self, document):
+        return self.label_template.render({"document": document})
+
+
+class DocumentMergeForm(forms.Form):
+    primary_document = DocumentChoiceField(
+        label="Primary record",
+        queryset=None,
+        help_text=(
+            "Select the document record to prioritize. "
+            "Combines all metadata into this document, adds the merged "
+            "documents into list of old PGP IDs."
+        ),
+        empty_label=None,
+        widget=forms.RadioSelect,
+    )
+    rationale = forms.CharField(
+        label="Rationale",
+        help_text="Provide a brief explanation for merging these records.",
+    )
+
+    def __init__(self, *args, **kwargs):
+        document_ids = kwargs.get("document_ids", [])
+        # ? : I don't understand what's going on here.
+        try:
+            del kwargs["document_ids"]
+        except KeyError:
+            pass
```

Comment:
looks like just deleting it from kwargs before we call the parent init method, because it otherwise would cause problems; you could uncomment to confirm and then add a better comment